File: frontend/src/app/core/interfaces/empleado.ts
import { Negocio } from './negocio';

export interface Empleado {
  id_empleado: number;
  nombre: string;
  dni: string;
  cargo: string;
  telefono: string;
  email: string;
  fecha_inicio: Date;
  id_negocio: number;
  negocio: Negocio;
}

File: backend/schemas/empleado.py
from pydantic import BaseModel
from datetime import datetime


class EmpleadoBase(BaseModel):
    nombre: str
    dni: int
    cargo: str
    telefono: str | None = None
    email: str | None = None
    id_negocio: int


class EmpleadoCreate(EmpleadoBase):
    """Esquema para crear empleado."""
    pass


class Empleado(EmpleadoBase):
    """Respuesta completa."""
    id: int
    fecha_inicio: datetime
    estado: bool

    model_config = {
        "from_attributes": True
    }

class EmpleadoSimple(BaseModel):
    nombre: str

    model_config = {
        "from_attributes": True
    }
